pennylane/labs/resource_estimation/templates/subroutines.py

```diff
@@ -348,5 +911,444 @@ def resource_params(self) -> dict:
 
     @classmethod
     def resource_rep(cls, cmpr_ops, num_ctrl_wires) -> CompressedResourceOp:
+        r"""Returns a compressed representation containing only the parameters of
+        the Operator that are needed to compute a resource estimation.
+
+        Args:
+            cmpr_ops (list[CompressedResourceOp]): The list of operators, in the compressed representation, corresponding to the unitaries of the LCU representation of the hamiltonian being qubitized.
+            num_ctrl_wires (int): The number of qubits used to prepare the coefficients vector of the LCU.
+
+        Returns:
+            CompressedResourceOp: the operator in a compressed representation
+        """
         params = {"cmpr_ops": cmpr_ops, "num_ctrl_wires": num_ctrl_wires}
         return CompressedResourceOp(cls, params)
+
+
+class ResourceQROM(qml.QROM, ResourceOperator):
```

Comment:
After looking at the code I would say that the QROM decomposition does not need the unary iterator version. You would simply have to add the RersourceSelect. The key is that ResourceSelect already has the two possible decompositions

Reply:
or the issue lies in the specification of the wires? 🙈